react/src/App.js

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+import BookList from './components/BookList';
+import BookFilters from './components/BookFilters';
+import './App.css';
+import 'roboto-npm-webfont';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {filterType: 0, filterText: ""};
+    this.handleTextFilter = this.handleTextFilter.bind(this);
+    this.handleFieldFilter = this.handleFieldFilter.bind(this);
+  }
```

Comment:
This `constructor` can be avoided by writing arrow functions and state as an object:
```
class App extends Component {
  state = { filterType: 0, filterText: '' };

  handleTextFilter = (filterText) => this.setState({ filterText });
  handleFieldFilter = (filterField) => this.setState({ filterField });

  render() {
    ...
  }
}
```